diff --git a/cmake/config-ix.cmake b/cmake/config-ix.cmake
index 7406fba..2f99b17 100644
--- a/native/src/external/libcxx/cmake/config-ix.cmake
+++ b/native/src/external/libcxx/cmake/config-ix.cmake
@@ -22,7 +22,7 @@ if (NOT LIBCXX_USE_COMPILER_RT)
     if(ANDROID)
       check_library_exists(gcc __gcc_personality_v0 "" LIBCXX_HAS_GCC_LIB)
     else()
-      check_library_exists(gcc_s __gcc_personality_v0 "" LIBCXX_HAS_GCC_S_LIB)
+      #check_library_exists(gcc_s __gcc_personality_v0 "" LIBCXX_HAS_GCC_S_LIB)
     endif()
   endif()
 endif()
-- 
2.43.0
